OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,297 @@
+                case 427:
+                    return IN_APP;
+                case 498:
+                    return WHATSAPP;
+                case 15:
+                    return MAIL;
+                case 495:
+                    return INBOX;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Initializes an instance of OptimovePreferenceCenter
+     *
+     * @param currentConfig current config
+     */
+    public static void initialize(OptimoveConfig currentConfig) {
+        shared = new OptimovePreferenceCenter();
+        config = currentConfig.getPreferenceCenterConfig();
+
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    /**
+     * Asynchronously runs preferences get handler on UI thread. Handler receives a single argument Preferences
+     *
+     * @param preferencesGetHandler handler
+     */
+    public void getPreferencesAsync(@NonNull PreferencesGetHandler preferencesGetHandler) {
+        UserInfo userInfo = Optimove.getInstance().getUserInfo();
+        String userId = userInfo.getUserId();
+
+        if (userId == null || Objects.equals(userId, userInfo.getVisitorId())) {
+            Log.w(TAG, "Customer ID is not set");
+            return;
+        }
+
+        Runnable task = new GetPreferencesRunnable(userId, preferencesGetHandler);
+        executorService.submit(task);
+    }
+
+    /**
+     * Asynchronously runs preferences set handler on UI thread. Handler receives a single Boolean result argument
+     *
+     * @param preferencesSetHandler handler
+     * @param updates               list of preference updates to set
+     */
+    public void setCustomerPreferencesAsync(@NonNull PreferencesSetHandler preferencesSetHandler, List<PreferenceUpdate> updates) {
+        UserInfo userInfo = Optimove.getInstance().getUserInfo();
+        String userId = userInfo.getUserId();
+
+        if (userId == null || Objects.equals(userId, userInfo.getVisitorId())) {
+            Log.w(TAG, "Customer ID is not set");
+            return;
+        }
+
+        Runnable task = new SetPreferencesRunnable(userId, preferencesSetHandler, updates);
+        executorService.submit(task);
+    }
+
+    static class GetPreferencesRunnable implements Runnable {
+        private final String customerId;
+        private final PreferencesGetHandler callback;
+
+        GetPreferencesRunnable(String customerId, PreferencesGetHandler callback) {
+            this.customerId = customerId;
+            this.callback = callback;
+        }
+
+        @Override
+        public void run() {
+            String region = config.getRegion();
```

Comment:
region here is dev/eu/us. but dev endpoint is still dev-pb. so

a) we still need mapping . region cannot be `dev-pb` -- public is not part of region, it's a random suffix in the domain
b) i asked Yuri if they could remove -pb . I already saw  bugs related to this

Reply:
this one is currently unused `private String mapRegion(String region) {`

we need to either delete or update depending on what Yuri says

Reply:
removed mapping now

Reply:
so, based on slack just remove mapRegion